### ISSUE TITLE
Metafiles with INTEGER Column Types

### ISSUE THREAD

**Issue by jpalmerpivotal** [imsweb/validation-translation#1]
Thanks for publishing this repo. Other stuff came up and I am finally able to get back to working on this task.

The demos are very appreciated, I was able to incorporate information from them to run a number of state metafiles successfully. I have run into two main difficulties: mapping state specific fields and Wisconsin actually did use `INTEGER` as a type. Additionally I had one concern that I am unclear if it is actually a problem. I will break them each into separate issues.
______

When I ran Wisconsin translation, I got the following error:
```
Exception in thread "main" java.lang.RuntimeException: Unsupported type for table 'EDITSCONFIG: INTEGER
```

After code digging I found this beautifully prescient comment:
```
// metafiles support "integer" but nobody use them; if somebody starts using them I will have to change the main translation to not add quotes...
if (!"CHAR".equals(result2.getMetaData().getColumnTypeName(i)) && !"VARCHAR".equals(result2.getMetaData().getColumnTypeName(i)))
```

So it seems like someone decided to use `INTEGER`. Not sure if anyone other than Wisconsin does but I figured this may be something you could knock out somewhat easily based on the comment.

**Comment by depryf** [comment]
I agree it looks like it will be simple to add support for it based on the comment. I will look into it.

**Comment by depryf** [comment]
This is more complex that I thought.

The problem is that this is not just a translation issue; the engine uses those tables/indexes and right now it assumes every values in those is a String. 

Supporting integers in the translation is not difficult (I actually already made the change).

But the engine also needs to be updated to properly handle integer values, and that is not a trivial change.

I thought about leaving those INTEGER as String, but that will introduce some subtle bugs because Integer and String don't compare the same way (and so "11" is going to be considered smaller than "2") and that will break the logic of the indexes.

Another solution is to leave those INTEGER as String, but compare the values as integers if they are all digit. That would be less work, but this doesn't feel right.

I am going to look into the engine and see how much effort it will take to properly support integers in tables and indexes.

**Comment by depryf** [comment]
@jpalmerpivotal would it be possible to see how that table looks like in GeneditsWriter? Maybe a screenshot of the headers and the first few rows? And maybe a sample of some logic that uses that table or index?

Things can get complicated. For example, what is the meaning of defining an index on two columns, one being a VARCHAR and one being an INTEGER? In that case the index has to be a VARCHAR, but that is going to create index entries that have different lengths, and that can't be properly compared to each other. 

I am sure the Genedits engine has a way to support all that, but it's a black box to me. And so to fix something like this, I need to see a real use-case and fully understand it.

If you prefer, you can email me directly about this.

**Comment by jpalmerpivotal** [comment]
So a strange thing happened. I was digging into the Wisconsin Metafile right now to try and get it working. I thought I would give you a link to the metafile source so you could play with it on your end. I went to the [clearinghouse](https://www.naaccr.org/standard-data-edits/?alp=&dt=#1628859400976-1924e77e-f744). The metafile I was using was provided internally from someone at the company and it was the v21.
When I ran the Wisconsin v22 at the clearinghouse the edits ran successfully and there was no integer issue.

I looked in v22 and it no longer has the table EditsConfig.

So I am thinking if I can run v22 edits I am probably set. Maybe it makes sense to deprioritize this again. If you still want to fix it I can email you the metafile that cause the issue, but it seems like we have returned to a state of `// but nobody uses them`.

Sorry that time got burned on this...

**Comment by depryf** [comment]
So technically the change I made did fix the translation and the "correct" tables/indexes will be created even with INTEGER columns.  

But fixing the engine to properly use those is not simple at all, and I would rather not attempt that, especially if it's not really needed. 

And so I think I am going to merge the changes I made and close this request. 

If you try to initialize the engine with a translated metafile that does contain INTEGER columns (that are actually used), then there will be an exception saying it's not supported. For all we know, that might never happen (it never did with the metafiles from the big standard setters like NAACCR).
